hls4ml/model/layers.py

```diff
@@ -1147,20 +1147,57 @@ class Resize(Layer):
     def initialize(self):
         inp = self.get_input_variable()
 
-        if self.get_attr('data_format') == 'channels_last':
-            if len(inp.shape) == 2:  # 1D -> width + chan
-                shape = [self.get_attr('out_width'), self.get_attr('n_chan')]
-                dims = [f'OUT_WIDTH_{self.index}', f'N_CHAN_{self.index}']
-            elif len(inp.shape) == 3:  # 2D -> height + width + chan
-                shape = [self.get_attr('out_height'), self.get_attr('out_width'), self.get_attr('n_chan')]
-                dims = [f'OUT_HEIGHT_{self.index}', f'OUT_WIDTH_{self.index}', f'N_CHAN_{self.index}']
+        if len(self.inputs) > 1:
+            # get the scales of Resize node from QONNX frontend
+            # see doc here: https://onnx.ai/onnx/operators/onnx__Resize.html
+            scales_idx = 2 if len(self.inputs) == 3 or len(self.inputs) == 4 else 1
```

Comment:
We don't currently handle the case where "sizes" is passed instead of a "scales". If "sizes" is passed, I think we will just get the exception, "Resize 1D and Resize 2D are the ones supported in hls4m", which is not appropriate. If "sizes" is passed, we need to either use it appropriately or raise an exception saying that we don't currently support "sizes." (Either choices are fine.)

Reply:
I added a check that raise an exception in case the input are 4, i.e. if sizes are passed. I think it is not needed to replicate the same check in this [optimiser](https://github.com/fastmachinelearning/hls4ml/pull/1122/files/b07e998d27de11c1334cacd0ea72b8b1a4b02efd#diff-ca5a48d22e8e2cad21ae235acb571aa79a9733cb64971063990099a43f39e996) since the code in layers.py is run first.

Reply:
Wait, didn't we say that self.inputs==2 was the old version ONNX when ROI was not used?

Reply:
Resize version 2 only has two inputs, the data input and scale. How old is it? I think it's easy to support.